app.rb

```diff
@@ -1,19 +1,81 @@
 require_relative "data"
+require_relative "models/apartment"
+require_relative "models/tenant"
 apartments = data[:apartments]
 tenants = data[:tenants]
+ruby_apartments = []
+ruby_tenants = []
+
+ruby_apartments = apartments.each do |id, address, rent, size|
+   ruby_apartments.push(Apartment.new(id, address, rent, size))
+end
+
+ruby_tenants = tenants.each do |id, name, age, apartment_id|
+   ruby_tenants.push(Tenant.new(id, name, age, apartment_id))
+end
+
+puts "Welcome to MyAppartmentManager.com"
+puts "1) To View All Apartments"
+puts "2) To View All Tenants"
+puts "3) To Exit"
+
+selection = gets.chomp.to_i
+
+if selection == 1
+  puts ruby_apartments
+elsif selection == 2
+  puts ruby_tenants
+else
+  puts "Thanks for using MyAppartmentManager.com!"
+  puts "If you would like to use our service again,"
+  puts "please reload this page."
+end
+
+
+
+
 
 # First, Open the data.rb an inspect the data. Identify and write, in comments, the following:
   # Explain how the data is structured
+  # The data is structured in 2 arrays, apartments and tenants, each consisting of multiple hashes.
   # What are the properties for each of the two types of hashes
+  # The hashes within the apartments array contain keys for id, address, monthly_rent, and square_feet.
+  # The hashes within the tenants array contain keys for id, name, age, and appartment_id.
 
 # Use enumerables to -
   # Print all the addresses for the apartments
+#   addresses = ruby_apartments.map do |apartment|
+#     apartment[:address]
+#   end
+#   puts addresses
+
   # Print all the names for tenants
+  # names = ruby_tenants.map do |name|
+  #   name[:name]
+  # end
+  # puts names
```

Comment:
This works, but I feel like our main goal here is the *side effect* of `puts`ing, and your approach creates this names array which isn't strictly necessary. Instead I'd suggest using `each` when your main goal is side effects:

```ruby
ruby_tenants.each do |tenant|
  puts tenant.name
end
```